src/plots/plots.js

```diff
@@ -2045,6 +2048,20 @@ plots.doAutoMargin = function(gd) {
         // now cycle through all the combinations of l and r
         // (and t and b) to find the required margins
 
+        // make sure that the reservedMargin is the minimum needed
+        for(var s in reservedMargins) {
+            var autoMarginPush = 0;
+            for(var m in pushMargin) {
+                if(m !== 'base') {
+                    if(isNumeric(pushMargin[m][s].size)) {
+                        autoMarginPush += pushMargin[m][s].size;
```

Comment:
This logic doesn't seem right to me. For one if you have multiple things pushing the same margin out from the same anchor point, you'd get the max, not the sum, wouldn't you? Also if the anchor isn't exactly the edge of the plot area (ie `val !== 0 or 1` depending on the side) you're not going to eat up the same amount of the reserved margin.

I would have thought, in fact, that what we really want to do is take the reserved margins out of ONLY `base`. The idea being that `base` margins are allowed to overlap the reserved margins, so instead of the originally-specified margin, just say how much of those margins we still want after taking out the reserved amount. If that's right, then we wouldn't need this new block at all, instead we'd change `pushMargin.base` [above](https://github.com/plotly/plotly.js/pull/6616/files#diff-bb59a2aa90fd14ef3218544010e7fe42f370e10f0c8ee702b3f85ab86887d970R2041) to:
```js
        pushMargin.base = {
            l: {val: 0, size: ml - reservedMargins.l},
            r: {val: 1, size: mr - reservedMargins.r},
            t: {val: 1, size: mt - reservedMargins.t},
            b: {val: 0, size: mb - reservedMargins.b}
        };
```

Reply:
Yes, I'm with you on the first point there about taking the max. 

A bit less clear on what behaviour you're anticipating when the anchor isn't 1 or 0... playing around with the current implementation and it accounts for the `reservedMargin` as I'd expect. I tested out your suggestion here and am still left with excess margin spacing (perhaps I need to subtract the `reservedMargin`s when `ml`, `mr`, etc are defined more directly). 

Could you demo or further describe the behaviour that the current logic doesn't accomplish?

Reply:
Noted this as a follow up in https://github.com/plotly/plotly.js/issues/6624